### PR TITLE
apps/server/server.cpp: extend server to use rgb sensor

### PR DESCRIPTION
**Pull request by SeptimiuVana**
Signed-off-by: SeptimiuVana <Vana.Septimiu@analog.com>

### CODE REVIEW

File: apps/server/server.cpp [hunk 1]
```diff
@@ -314,13 +320,16 @@ void invoke_sdk_api(payload::ClientRequest buff_recv) {
...
-        aditof::SensorDetails depthSensorDetails;
-        camDepthSensor->getDetails(depthSensorDetails);
+
+        int depth_sensor_id = 0;
         auto pbSensorsInfo = buff_send.mutable_sensors_info();
-        sensorV4lBufAccess =
-            std::dynamic_pointer_cast<aditof::V4lBufferAccessInterface>(
-                camDepthSensor);
+        for (const auto &depthSensor : depthSensors) {
+            camDepthSensor[depth_sensor_id] = depthSensor;
```
Comment (dNechita) [inline apps/server/server.cpp:327]: depth_sensor_id should be incremented in this loop. Otherwise it will always be 0.

Comment (SeptimiuVana) [reply apps/server/server.cpp:327]: yes, thank you! fixed

File: apps/server/buffer.proto [hunk 2]
```diff
@@ -58,6 +58,7 @@ message ClientRequest
   repeated bytes func_bytes_param = 40; // List of function parameters of type byte-array
   bool expect_reply = 50;               // Whether a response with data is expected or not
   FrameDetails frame_type = 60;         // Frame type information
+  int32 id = 70;
```
Comment (dNechita) [inline apps/server/buffer.proto:61]: It would make more sense to follow the structure of the other sensors (temperature, storage).
I mean have another message called ImagerInfo with fields name and id. And add it to the SensorsInfo message.

File: sdk/include/aditof/sensor_definitions.h [hunk 3]
```diff
@@ -45,12 +45,23 @@ namespace aditof {
  * @struct SensorDetails
  * @brief Provides details about the device
  */
-struct SensorDetails {
+struct SensorInformations {
```
Comment (dNechita) [inline sdk/include/aditof/sensor_definitions.h:48]: This seems to be used internally only. For this reason it shouldn't be public. Also you could have directly 2 members (id and name) instead of this struct.

File: apps/server/buffer.proto [hunk 4]
```diff
@@ -34,6 +40,7 @@ message SensorsInfo
 {
   repeated StorageInfo storages = 10;               // Information about available storages
   repeated TemperatureSensorInfo temp_sensors = 20; // Information about available temperature sensors
+  repeated ImagerInfo image_sensors = 30;                    // Information about available image sensors
```
Comment (dNechita) [inline apps/server/buffer.proto:43]: would be nice to align the comment with the rest of comments

File: apps/server/buffer.proto [hunk 5]
```diff
@@ -58,6 +65,7 @@ message ClientRequest
   repeated bytes func_bytes_param = 40; // List of function parameters of type byte-array
   bool expect_reply = 50;               // Whether a response with data is expected or not
   FrameDetails frame_type = 60;         // Frame type information
+  SensorsInfo sensors_info = 70;        // List of informations regarding which sensor to address
```
Comment (dNechita) [inline apps/server/buffer.proto:68]: maybe reword in a simpler way: "The sensor to address" or "The sensor for which the request should apply"
